tools/dashboard/generate_dashboard.py

```diff
@@ -183,19 +269,21 @@ def gen_archive(config, log, outdir):
             }
 
         # read all archived reports
-        archive_reports = dict()
+        archive_reports: Dict[GitSha, Report] = {}
         for fn in archive_files:
+            report: Report
             if fn in reports_cache:
                 report = reports_cache[fn]
             else:
                 with gzip.open(fn, "rb") as f:
                     report_txt = f.read().decode("utf-8", errors="replace")
                 report = parse_report(report_txt, log)
-                report["url"] = path.basename(fn)[:-3]
+                report.archive_path = path.basename(fn)[:-3]
                 reports_cache[fn] = report
-            sha = report["git-sha"]
-            assert sha not in archive_reports
-            archive_reports[sha] = report
+            if not report.sha:
+                print(report)
+            assert report.sha and report.sha not in archive_reports
```

Comment:
using `assert ..., "some message here"` helps to document the assertion and giving additional feedback in case it fails